src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -55,13 +56,19 @@ OtpUser preCreateHook(OtpUser user, Request req) {
             Auth0Connection.ensureApiUserHasApiKey(req);
             user.applicationId = requestingUser.apiUser.id;
         }
-        user.mobilityMode = calculateMobilityMode(user);
+        if (user.mobilityProfile == null) {
+            user.mobilityProfile = new MobilityProfile();
```

Comment:
I did that for backward compatibility, since our existing OtpUser db instances don't have `mobilityProfile` and it might be null.  I considered doing it inside `calculateMobilityMode()` but that would make it a side-effect.

Reply:
My thought was that `mobilityProfile` should stay null, especially for deployments that don't use it. Now, if `mobilityProfile` is specified, go ahead and `calculateMobilityMode()`.

Reply:
Have refactored as requested below.